lib/tokenize.cpp

```diff
@@ -3108,6 +3108,16 @@ void Tokenizer::simplifyArrayAccessSyntax()
     }
 }
 
+void Tokenizer::simplifyParameterVoid()
+{
+    for (Token* tok = list.front(); tok; tok = tok->next()) {
```

Comment:
sorry .. it seems this simplification will be used in C code also? In C code the `void` should be kept. In C++ it doesn't matter but in C it is important.

Reply:
There are too many places in the code where we assume a `void` parameter function is a unary function(not just in the templatesimplifier), which affects both C and C++ analysis. Its also difficult to ensure enough test coverage for these cases. Its best for the simplifications to canonicalize the source tokens as much as possible and have them link back to the raw tokens for checkers that need more details.